Purchases/Purchasing/StoreKit1/ProductsFetcherSK1.swift

```diff
@@ -54,6 +54,24 @@ class ProductsFetcherSK1: NSObject {
             request.start()
         }
     }
+
+    func products(withIdentifiers identifiers: Set<String>,
+                  completion: @escaping (Set<ProductDetails>) -> Void) {
+        self.sk1Products(withIdentifiers: identifiers) { skProducts in
+            let wrappedProductsArray = skProducts.map { SK1ProductDetails(sk1Product: $0) }
+            completion(Set(wrappedProductsArray))
+        }
+    }
+
+    @available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *)
+    func products(withIdentifiers identifiers: Set<String>) async -> Set<ProductDetails> {
```

Comment:
Do we need this function? It would never be called right? In this case we would be fetching SK2Products

Reply:
I wrote this one before you'd done the same in intro trial eligiblity PR, I guess it became irrelevant after the rebase and I'd missed it, thanks!

Reply:
scratch that, I thought this was somethign else. Yeah, this one doesn't make sense

Reply:
lol now I remember this one. I added it to make tests cleaner, in particular here: 
https://github.com/RevenueCat/purchases-ios/pull/832/files/e101209cf1fbd6c9c7ec06236111e42155744551#diff-797c8f10d3af321c61b1839a05121e7413747945a3770ee22ac71a38a379b41fR31